### PR TITLE
Fix issues #205, #206 and #216

### LINKED ISSUE #205: s-pipes-module-creator-maven-plugin Generates Wrong Module RDF Description
Fix generation of RDF description from java class with annotations `@SPipesModule` and `@Parameter` .

## Reproducing the bug
When compiling `s-pipes-modules` the plugin `s-pipes-module-creator-maven-plugin` generates wrong RDF description from s-pipes modules defined in classes. The specific symptom is:
* `tabular-module` has the same properties as defined by `text-analysis-module`

### PR DESCRIPTION
**Pull request by kostobog**
Fixes #205, #206 and #216

#205
- delete incorrect sharing of parameter declaration among modules in writeConstraintsToModel
- set constraint predicate as IRI instead of string
- add subClassOf triple to modules extending other modules, including abstract ones, to share parameter declaratoin with their super classes.
- Note that tools using the generated descriptions of modules that extend other modules will have to query the super classes for their parameter declaration.

#206
- Add SPipesModule and Param

#216 - resolved by commit [ae5869a](https://github.com/kbss-cvut/s-pipes/pull/210/commits/ae5869a9607d6245f0722c4bbd1cb944f85616be)

### CODE REVIEW

**Comment by kostobog** [comment]
Review notes:
- Abstract classes should not be converted to modules.
- specify comments of parameters from javadoc and `s-pipes-modules` project.
- this should be copied in all-modules.ttl do not include triples where the resource is the ontology. See module `module-text-analysis.ttl` in resource folder of `moodule-text-analysis` .

File: s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java [hunk 1]
```diff
@@ -34,6 +39,7 @@ public class DatasetDiscoveryModule extends AbstractModule {
      * URL of the Sesame server.
      */
     private static final Property P_USER_INPUT = getParameter("prp-user-input");
+    @Parameter(urlPrefix = TYPE_URI + "/", name = "prp-user-inpu")
```
Comment (blcham) [inline s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java:42]: ?

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructFormMetadataModule.java [hunk 2]
```diff
@@ -21,6 +22,7 @@
 /**
  * Compute form:has-origin-path and form:has-origin-path-id properties.
  */
+@SPipesModule(label = "construct form metadata", comment = "Compute form:has-origin-path and form:has-origin-path-id properties.")
```
Comment (blcham) [inline s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructFormMetadataModule.java:25]: Duplicate comment -- leave it only in the annotation.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java [hunk 3]
```diff
@@ -25,6 +26,10 @@
  * For input Q&A models constructs textual view of specified questions. Each textual view represent the question
  * and its sub-questions recursively.
  */
+@SPipesModule(label = "construct textual view", comment =
```
Comment (blcham) [inline s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java:29]: Duplicate comment.

Just note, in JAVA 15 we will use text blocks to make it more readable, see https://www.baeldung.com/java-multiline-string

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/FetchPossibleValuesModule.java [hunk 4]
```diff
@@ -19,6 +20,10 @@
  * Inputs are forms using Q&A model. Possible values of questions are added to questions that does not have
  * any value attached and contains possible value query.
  */
+@SPipesModule(label = "fetch possible values", comment =
```
Comment (blcham) [inline s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/FetchPossibleValuesModule.java:23]: duplicate comment

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/MergeFormMetadataModule.java [hunk 5]
```diff
@@ -20,6 +21,11 @@
  * question origin combined with executionId. New question instances are created using questionInstanceTemplate
  * which defaults to "doc:question-{_questionOriginHash}-{_executionId}".
  */
+@SPipesModule(label = "merge form metadata", comment =
```
Comment (blcham) [inline s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/MergeFormMetadataModule.java:24]: duplicate comment

File: s-pipes-modules/module-rdf4j/src/main/java/cz/cvut/spipes/modules/Rdf4jDeployModule.java [hunk 6]
```diff
@@ -35,12 +36,16 @@
  * into default context of repository (if p-rdf4j-context-iri is not specified)
  * or concrete context (if p-rdf4j-context-iri is specified).
  */
+@SPipesModule(label = "deploy", comment =
```
Comment (blcham) [inline s-pipes-modules/module-rdf4j/src/main/java/cz/cvut/spipes/modules/Rdf4jDeployModule.java:39]: duplicate comment

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/RDF2CSVModule.java [hunk 7]
```diff
@@ -29,6 +30,9 @@
  * The table is constructed from column and row resources defined in TableSchema and saves it as a new CSV file.
  * </p>
  */
+@SPipesModule(label = "RDF2CSV", comment = "Module for converting RDF (representing table) to CSV. " +
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/RDF2CSVModule.java:33]: duplicate comment

**Review by blcham — CHANGES_REQUESTED** [review]
Did not go over it extensively yet.

File: s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java (hunk 1 above)
Comment (kostobog) [reply s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java:42]: Not sure, what is wrong here. Can you elaborate?

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/RDF2CSVModule.java (hunk 7 above)
Comment (kostobog) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/RDF2CSVModule.java:33]: done.

File: s-pipes-modules/module-rdf4j/src/main/java/cz/cvut/spipes/modules/Rdf4jDeployModule.java (hunk 6 above)
Comment (kostobog) [reply s-pipes-modules/module-rdf4j/src/main/java/cz/cvut/spipes/modules/Rdf4jDeployModule.java:34]: done.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/MergeFormMetadataModule.java (hunk 5 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/MergeFormMetadataModule.java:24]: done.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/FetchPossibleValuesModule.java (hunk 4 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/FetchPossibleValuesModule.java:23]: done.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java (hunk 3 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java:29]: done.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructFormMetadataModule.java (hunk 2 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructFormMetadataModule.java:25]: done.

File: s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java (hunk 1 above)
Comment (kostobog) [reply s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java]: done.

Comment (blcham) [reply s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java]: name = "prp-user-inpu" vs. getParameter("prp-user-input")

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructFormMetadataModule.java (hunk 2 above)
Comment (blcham) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructFormMetadataModule.java:25]: @kostobog  I do not see where it was done.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java (hunk 3 above)
Comment (blcham) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java:29]: not done yet

**Comment by blcham** [comment]
@kostobog please create separate ticket for https://github.com/kbss-cvut/s-pipes/pull/210/commits/7152270bee474eded021edeed438d8d5b6b614a6

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java (hunk 3 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java:25]: Should be done now.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/FetchPossibleValuesModule.java (hunk 4 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/FetchPossibleValuesModule.java:19]: Should be done now.

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/MergeFormMetadataModule.java (hunk 5 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/MergeFormMetadataModule.java:19]: Should be done now.

File: s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java (hunk 1 above)
Comment (kostobog) [reply s-pipes-modules/module-dataset-discovery/src/main/java/cz/cvut/spipes/modules/DatasetDiscoveryModule.java]: done

File: s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java (hunk 3 above)
Comment (kostobog) [reply s-pipes-modules/module-form/src/main/java/cz/cvut/spipes/modules/ConstructTextualViewModule.java:25]: Should be done now.

**Comment by kostobog** [comment]
Javadoc comments are moved into annotations.
Copying manually managed ontology with module descriptions from root of resource folder to `all-modules.ttl` implemented. 
The output `all-modules.ttl` is formatted using frames.

File: s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/src/main/java/RdfAnnotationProcessorMojo.java [hunk 8]
```diff
@@ -203,10 +217,45 @@ private Set<Class<?>> readAllModuleClasses(MavenProject project) throws Malforme
         return moduleClasses;
     }
 
+    /**
+     * Reads manually managed modules ontology from maven sub-project.
+     * The ontology is loaded from the resource folder of the sub-project and it is expected to start with the artifact
+     * id of the sub-project followed by the postfix ".sms.tll". For example, assume the input parameter project
```
[the remark this answers is not in the document]
Comment (blcham) [reply s-pipes-modules-utils/s-pipes-module-creator-maven-plugin/src/main/java/RdfAnnotationProcessorMojo.java]: @kostobog FYI, resolve when you read the comment

**Pull request merged**